File: src/main/java/me/lokka30/levelledmobs/util/MicroLogger.java
package me.lokka30.levelledmobs.util;

import java.util.logging.Logger;

import me.lokka30.levelledmobs.LevelledMobs;
import org.bukkit.Bukkit;
import org.bukkit.ChatColor;

@SuppressWarnings("deprecation")
public class MicroLogger {

    final boolean serverIsSpigot;
    private final Logger logger;
    private String prefix;

    /**
     * Create a new instance of MicroLogger with a custom prefix
     *
     * @param prefix the prefix to use
     * @author lokka30
     * @since unknown
     */
    public MicroLogger(String prefix) {
        this.prefix = prefix;
        this.logger = Bukkit.getLogger();
        this.serverIsSpigot = LevelledMobs.getInstance().getVerInfo().getIsRunningSpigot();
    }

    /**
     * @return the current prefix
     * @author lokka30
     * @since unknown
     */
    public String getPrefix() {
        return prefix;
    }

    /**
     * @param prefix the prefix that should be set
     * @author lokka30
     * @since unknown
     */
    public void setPrefix(String prefix) {
        this.prefix = prefix;
    }

    /**
     * @param message message to send with INFO log-level
     * @author lokka30, stumper66
     * @since unknown
     */
    public void info(String message) {
        if (serverIsSpigot)
            Bukkit.getServer().getConsoleSender().sendMessage(MessageUtils.colorizeAll(prefix + message));
        else
            logger.info(MessageUtils.colorizeAll(prefix + message));
    }

    /**
     * @param message message to send with WARNING log-level
     * @author lokka30, stumper66
     * @since unknown
     */
    public void warning(String message) {
        if (serverIsSpigot)
            Bukkit.getServer().getConsoleSender().sendMessage(MessageUtils.colorizeAll(ChatColor.YELLOW + "[WARN] " + ChatColor.RESET + prefix + message));
        else
            logger.warning(MessageUtils.colorizeAll(prefix + message));
    }

    /**
     * @param message message to send with ERROR log-level
     * @author lokka30, stumper66
     * @since unknown
     */
    public void error(String message) {
        if (serverIsSpigot)
            Bukkit.getServer().getConsoleSender().sendMessage(MessageUtils.colorizeAll(ChatColor.RED + "[ERROR] " + ChatColor.RESET + prefix + message));
        else
            logger.severe(MessageUtils.colorizeAll(prefix + message));
    }
}

File: levelledmobs-plugin/src/main/kotlin/io/github/arcaneplugins/levelledmobs/util/MicroLogger.kt
@file:Suppress("DEPRECATION")

package io.github.arcaneplugins.levelledmobs.util

import java.util.logging.Logger
import io.github.arcaneplugins.levelledmobs.LevelledMobs
import org.bukkit.Bukkit
import org.bukkit.ChatColor

/**
 * Logging utility class.
 *
 * @author lokka30, stumper66
 * @since 1.0.3
 */
@SuppressWarnings("deprecation")
class MicroLogger(
    private val prefix: String
) {
    private val serverIsSpigot: Boolean = LevelledMobs.instance.ver.isRunningSpigot
    private val logger: Logger = Bukkit.getLogger()

    /**
     * @param message message to send with INFO log-level
     * @author lokka30, stumper66
     * @since unknown
     */
    fun info(message: String) {
        if (serverIsSpigot) Bukkit.getServer().consoleSender.sendMessage(MessageUtils.colorizeAll(prefix + message))
        else logger.info(MessageUtils.colorizeAll(prefix + message))
    }

    /**
     * @param message message to send with WARNING log-level
     * @author lokka30, stumper66
     * @since unknown
     */
    fun warning(message: String) {
        if (serverIsSpigot) Bukkit.getServer().consoleSender.sendMessage(MessageUtils.colorizeAll(ChatColor.YELLOW.toString() + "[WARN] " + ChatColor.RESET + prefix + message))
        else logger.warning(MessageUtils.colorizeAll(prefix + message))
    }

    /**
     * @param message message to send with ERROR log-level
     * @author lokka30, stumper66
     * @since unknown
     */
    fun error(message: String) {
        if (serverIsSpigot) Bukkit.getServer().consoleSender.sendMessage(MessageUtils.colorizeAll(ChatColor.RED.toString() + "[ERROR] " + ChatColor.RESET + prefix + message))
        else logger.severe(MessageUtils.colorizeAll(prefix + message))
    }
}